src/include/binder/bind_node_visitor.h

```diff
@@ -57,6 +58,7 @@ class BindNodeVisitor : public SqlNodeVisitor {
   void Visit(expression::CaseExpression *expr) override;
   // void Visit(const expression::ConstantValueExpression *expr) override;
   void Visit(expression::TupleValueExpression *expr) override;
+  concurrency::Transaction *txn = nullptr;
```

Comment:
First, it is thread-safe. Second, in bind_node_visitor, it defines many Visit function and there are only two Visit function use txn, however, the Visit functions is recursive, so pass in as an argument will change binder_node_visitor a lot for example, use if..else.. at beginning of every Visit function to see if this node need txn. Compared with pass in as an argument, member variable won't change code so much and brief.